tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,108 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package reset_partition;
+# Summary: reset_partition: Ensure and reset hard disk partition for virt_autotest.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+
+sub reset_partition {
+    my $self       = shift;
+    my $new_disk   = "";
+    my $virt_point = "/var/lib/libvirt";
+
+    # get some debug info about hard disk topology
+    record_info("report file system disk space", script_output("df"));
+    record_info("find a filesystem",             script_output("findmnt"));
+    record_info("find a filesystem",             script_output("ls -al /var/lib/libvirt"));
+    script_run 'df --output=avail /var/lib/libvirt';
+
+    my ($ACTIVE_DISK_NAME, $AVAILABLE_DISK_SIZE) = $self->get_active_disk_and_available_space();
+    # Disk which /var/lib/libvirt resides on
+    record_info('Detect Active DISK NAME:', $ACTIVE_DISK_NAME);
+    # Ensure available disk space under /var/lib/libvirt
+    record_info('Detect Available DISK SIZE:', $AVAILABLE_DISK_SIZE . 'GiB');
+    # Define the expected disk space for /var/lib/libvirt
+    my $expected_disk_size = (get_var("XEN")) ? '60' : '40';
+
+    # debug info
+    record_info('Excepted Disk Size:', $expected_disk_size);
+
+    if ($ACTIVE_DISK_NAME eq 'sda') {
+        $new_disk = "/dev/sdb";
+    } else {
+        $new_disk = "/dev/sda";
+    }
```

Comment:
Good way! But better check whether the disks are named with sd[x] way only and sda/sdb exists actually.

Reply:
Good point, will add the check whether the disks are named with sd[x] way only and sda/sdb exists actually.

Reply:
Had added the check point whether the disks are named with sd[x] way only and sda/sdb exists actually.

Reply:
@guoxuguang Where is the checking? Sorry I do not find it.

Reply:
Please refer line 38 - 42 code to check whether the disks are named with sd[x] way only and sda/sdb exists actually
```
    my @disks = qw/sda sdb sdc/;
    my $dev   = "/dev/";
    foreach my $disk (@disks) {
        assert_script_run('file ' . $dev . $disk);
    }
```